flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/OperatorState.java

```diff
@@ -41,6 +41,18 @@
  */
 public class OperatorState implements CompositeStateHandle {
 
+    private static final int FULLY_FINISHED_PLACEHOLDER_INDEX = -1;
+
+    /** A special content that marks this operator as fully finished. */
+    private static final Map<Integer, OperatorSubtaskState> FULLY_FINISHED_STATE =
+            new HashMap<Integer, OperatorSubtaskState>() {
+                {
+                    this.put(
+                            FULLY_FINISHED_PLACEHOLDER_INDEX,
+                            OperatorSubtaskState.builder().build());
+                }
+            };
```

Comment:
Why not `Collections.singletonMap(FULLY_FINISHED_PLACEHOLDER_INDEX, OperatorSubtaskState.builder().build()));`?

Reply:
Very thanks, this is exactly I'm looking for at first, but I do not found it...  :-)